### ISSUE TITLE
Prisma executes transaction twice

### ISSUE THREAD

**Issue by Hazeman28** [prisma/prisma#11686]
### Bug description

After upgrading to Prisma version `3.9.1` from version `3.8.1`, Prisma started to execute transactions twice if used in a non-async function.

For example, if you have a GraphQL resolver written like this:
```typescript
t.field("createBanner", {
  type: "Banner",
  args: {
    input: nonNull("BannerInput"),
  },
  resolve: (_, { input }, context) => {
    assertAccess(context, ["ADMIN"]);

    return prisma.banner.create({
      data: input,
    });
  },
});
```
Prisma will execute the create operation twice, resulting in duplicate entries in the database. But if rewrite this resolver with a simple change of adding `async`, thus making the return value an explicit `Promise`, then issue is resolved:
```diff
t.field("createBanner", {
  type: "Banner",
  args: {
    input: nonNull("BannerInput"),
  },
-  resolve: (_, { input }, context) => {
+  resolve: async (_, { input }, context) => {
    assertAccess(context, ["ADMIN"]);

    return prisma.banner.create({
      data: input,
    });
  },
});
```

### How to reproduce

Use Prisma inside of a non-async function.

[I've also recorded a reproduction of this bug](https://youtu.be/D4QD6DJLbTU). In the video recording, besides the aforementioned issue, I've encountered an odd problem, where I had some sort of a ghost node task hanging in the background, preventing me from restarting the dev server, but this might be unrelated to Prisma.

### Prisma information

Relevant part of the `schema.prisma`

<details>
<summary>Prisma schema</summary>

```prisma
model Banner {
  id          Int        @id @default(autoincrement())
  createdAt   DateTime   @default(now()) @map("created_at") @db.Timestamptz(6)
  updatedAt   DateTime   @default(now()) @updatedAt @map("updated_at") @db.Timestamptz(6)
  image       String
  mobileImage String?    @map("mobile_image")
  note        String?
  pathMatch   String     @map("path_match")
  exact       Boolean    @default(false)
  link        String?
  type        BannerType @default(TOP)
  validUntil  DateTime?  @map("valid_until") @db.Timestamptz(6)

  @@map("banners")
}
```

</details>

### Environment & setup

- OS: macOS Monterey 12.2
- Database: PostgreSQL
- Node.js version: 16.13.1

### Prisma Version

```
prisma                  : 3.9.1
@prisma/client          : 3.9.1
Current platform        : darwin-arm64
Query Engine (Node-API) : libquery-engine bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/libquery_engine-darwin-arm64.dylib.node)
Migration Engine        : migration-engine-cli bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/migration-engine-darwin-arm64)
Introspection Engine    : introspection-core bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/introspection-engine-darwin-arm64)
Format Binary           : prisma-fmt bcc2ff906db47790ee902e7bbc76d7ffb1893009 (at node_modules/@prisma/engines/prisma-fmt-darwin-arm64)
Default Engines Hash    : bcc2ff906db47790ee902e7bbc76d7ffb1893009
Studio                  : 0.457.0
```

**Comment by gopidon** [comment]
Breaking issue. Reproducible.

The following code creates duplicate posts without the async resolve function as stated above. If I use ays', things are back to normal. 

t.field('createPost', {
      type: 'Post',
      args: {
        data: arg({
          type: "PostCreateInput",
        })
      },
      resolve:  (parent, args, ctx) => {
        //getUser(ctx.req);
        //console.log("Calling post create!", args.data)
        return ctx.prisma.post.create({
          data: args.data
        })
      }
    }),

**Comment by torohaifisch** [comment]
found this bug too after upgrading to 3.9.1 but with normal operations,

using this:

      createNotaryPin: (_parent: any, args: any, ctx: any) => {

      args.code = `N-${v4()}`;

      return ctx.prisma.notaryPin.create({ data: args });
    }

executes the create operation twice or even three times.

**Comment by millsp** [comment]
Are you using `interactiveTransactions` feature flag?

**Comment by millsp** [comment]
@Hazeman28 @torohaifisch could you please provide a reproduction?

**Comment by torohaifisch** [comment]
> @Hazeman28 @torohaifisch could you please provide a reproduction? I was not able to reproduce this.

i will try and get a working repro showing the error. I downgraded to 3.8.1 and everything works fine

**Comment by torohaifisch** [comment]
here is an example project:
https://github.com/torohaifisch/prismaError/tree/master

project runs apollo sandbox to execute graphql queries, execute a createObservation mutation and it will sometimes execute the create operation twice.
You can verify this if you run the query operation getObservationList, it will show more registers than the times you executed the mutations. Another way to notice is that you wont get continuous ids on the mutation response.
This doesnt happen if i downgrade prisma and prisma client to 3.8.1

**Comment by Hazeman28** [comment]
> Are you using `interactiveTransactions` feature flag?

@millsp no, I'm not using any feature flags.

**Comment by millsp** [comment]
Thanks for the clear reproduction, I can confirm this. Apollo is doing a double call to `then` on the `PrismaPromise` which causes it to execute twice. We used to have a mechanism to prevent double execution, but that is gone. I'll push a fix later today.

**Comment by millsp** [comment]
Hey, the fix for this issue has been released via `3.9.2`. Thank you for reporting this issue!
